### ISSUE TITLE
No module named 'openai'

### ISSUE THREAD

**Issue by psociety** [RedisVentures/LLM-Document-Chat#6]
Hi!
Thank you for creating this project!

I've tried to use it but i'm running on to the following error:
1. did git clone of the repo
2. docker-compose up inside `docker/local`
3. Visited http://127.0.0.1:8888/lab/tree/redis-llm-doc-chat.ipynb
4. Executed first step, everything ok, it asked for restart, so i did `docker-compose stop` / `docker-compose start`
5. When running the second step i get `ModuleNotFoundError: No module named 'openai'`
![Screenshot from 2023-06-28 14-51-06](https://github.com/RedisVentures/LLM-Document-Chat/assets/29523682/685a7368-341b-4aa3-9fd7-75d78884bb5f)

If i check the first step it is true that it's not installing anything related to openai:
`%pip install redis pypdf PyPDF2 python-dotenv transformers tiktoken llama_index==0.6.8`
Is it missing some dependency?

If i change the first step so it also installs `pip install openai`, i just get a different error on step 2:
`
TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
`

If i install `pip install ipywidgets` this error disappears and i can continue.
Until i arrive on redis step where i get:
`ConnectionError: Error 99 connecting to localhost:6379. Cannot assign requested address.`

I don't understand the error as the redis container is running fine on that port:
![Screenshot from 2023-06-28 15-01-42](https://github.com/RedisVentures/LLM-Document-Chat/assets/29523682/263c89bb-8afd-4673-8e09-1a1f4c4d3111)

**Comment by antonum** [comment]
```
pip install llama_index
```
Would install OpenAI library among other things. There should be no need for a separate `pip install openai`, but you indeed need one for `ipywidgets`

Here is the `pip install` combination that works for me:
```
%pip install redis pypdf PyPDF2 python-dotenv transformers tiktoken ipywidgets llama_index==0.8.26
```
Note llama_index version bump.

There are some changes in Llama_index API that need to be accounted for.

Chunk Overlap is now float between 0 and 1:
```
# set maximum chunk overlap
max_chunk_overlap = float(os.getenv("CHUNK_OVERLAP"))
```
If you are using Redis Slack within the docker-compose setup it should be referred as: `redis`, instead of `localhost`

.env file
```
...
CHUNK_OVERLAP=0.2
REDIS_ADDRESS=redis
```

I'll submit these changes as a PR later.

**Comment by Spartee** [comment]
Addressed by #7. please comment or re-open if not the case
